get_training_set.py

```diff
@@ -0,0 +1,115 @@
+import xml.etree.ElementTree as xmlTree
+import numpy as np
+import os
+import tensorflow as tf
+from tqdm import tqdm
+
+class_names = ['car', 'watcher', 'base', 'armor_red', 'armor_blue']
+
+
+@tf.function
+def int64_feature(value):
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+
+def int64_list_feature(value):
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=value))
+
+
+def bytes_feature(value):
+    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))
+
+
+def bytes_list_feature(value):
+    return tf.train.Feature(bytes_list=tf.train.BytesList(value=value))
+
+
+def float_feature(value):
+    return tf.train.Feature(bytes_list=tf.train.FloatList(value=[value]))
+
+
+def float_list_feature(value):
+    return tf.train.Feature(float_list=tf.train.FloatList(value=value))
+
+
+def convert(size, box):
+    x1 = box[0] / size[0]
+    y1 = box[1] / size[1]
+    x2 = box[2] / size[0]
+    y2 = box[3] / size[1]
+    return [x1, y1, x2, y2]
```

Comment:
https://github.com/zzh8829/yolov3-tf2#nan-loss--training-failed--doesnt-converge
The input format required by original project needs the **normalized bboxes**.

Reply:
So data pipeline usually looks like `parse from tfrecord -> encoder -> model`. The normalization should be implemented in the `encoder` stage per network architecture. As other network than yolov3 could have different normalization strategies.

Reply:
```
def create_tf_example(example):
  # TODO(user): Populate the following variables from your example.
  height = None # Image height
  width = None # Image width
  filename = None # Filename of the image. Empty if image is not from file
  encoded_image_data = None # Encoded image bytes
  image_format = None # b'jpeg' or b'png'

  xmins = [] # List of normalized left x coordinates in bounding box (1 per box)
  xmaxs = [] # List of normalized right x coordinates in bounding box
             # (1 per box)
  ymins = [] # List of normalized top y coordinates in bounding box (1 per box)
  ymaxs = [] # List of normalized bottom y coordinates in bounding box
             # (1 per box)
  classes_text = [] # List of string class name of bounding box (1 per box)
  classes = [] # List of integer class id of bounding box (1 per box)

  tf_example = tf.train.Example(features=tf.train.Features(feature={
      'image/height': dataset_util.int64_feature(height),
      'image/width': dataset_util.int64_feature(width),
      'image/filename': dataset_util.bytes_feature(filename),
      'image/source_id': dataset_util.bytes_feature(filename),
      'image/encoded': dataset_util.bytes_feature(encoded_image_data),
      'image/format': dataset_util.bytes_feature(image_format),
      'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
      'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
      'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
      'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
      'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
      'image/object/class/label': dataset_util.int64_list_feature(classes),
  }))
  return tf_example
```
`xmins = [] # List of normalized left x coordinates in bounding box (1 per box)`
Maybe even Google prefers **normalized bboxes**.

Reply:
Google codebase is not really one of the clean ones. I don't think copying this style is a good choice.

Reply:
The other reason is that I still hesitate to rewrite the receiver code in the original project, since I do not understand how it works in parallel style. I can write a simple receiver with `for` loops, but it is not recommanded even for myself. I need some one to do this job. If so, I can change the convertor to meet the receiver's requirements, or I have to meet the original one's requirements.